src/main/ccl/cclut_utils.inc

```diff
@@ -28,13 +31,56 @@ subroutine public::cclutIsEmpty(cclutText)
     return (evaluate(textlen(trim(cclutText, 3)), 0, TRUE, FALSE))
 end ;cclutIsEmpty
 
+
+/**
+Returns the current revision of CCL. Allows CURREV to be mocked.
+**/
+subroutine public::cclutGetCurrentRevision(null)
+    return(currev)
+end ;cclutGetCurrentRevision
+
+
+/**
+Returns the current version of CCL. Allows CURCCLVER to be mocked.
+*/
+subroutine public::cclutGetCurrentVersion(null)
+    return(curcclver)
+end ;cclutGetCurrentVersion
+
+
 /**
-Invokes call parser on a specified string. Allows for overriding calls to call parser.
+Returns the group for the current CCL user. Allows CURGROUP to be mocked.
+*/
+subroutine public::cclutGetCurrentGroup(null)
+    return(curgroup)
+end ;cclutGetCurrentGroup
 
-@param cclutParserString
-    The string to invoke call parser on.        
+
+/**
+    Parses a delimeted string placing the (right trimmed) parts into the qual[*].str member of a provided record structure.
+    If the minimum requied CCL version is bumped above 8.13.2 / 9.1.2, arraySplit can be used and this can be removed.
 */
-subroutine public::cclutInvokeParser(cclutParserString)
-    call parser(cclutParserString)
-end ;cclutInvokeParser
+subroutine public::cclutDoArraySplit(cclutRec, cclutSource, cclutDelimiter)
+    declare cclutStat = i4 with protect, noconstant(0)
+    declare cclutDelimPos = i4 with protect, noconstant(0)
+    declare cclutDelimEnd = i4 with protect, noconstant(0)
+    declare cclutItemCount = i4 with protect, noconstant(0)
+
+    declare cclutDelimLen = i4 with protect, constant(textlen(cclutDelimiter))
+
+    set cclutDelimPos = findString(cclutDelimiter, cclutSource)
+    while(cclutDelimPos > 0 and cclutItemCount < 100)
```

Comment:
As unlikely as it is, do we need to document now that we're putting a cap on the number of columns that a user can mock in their table?

Reply:
lingering debug code to prevent an infinite loop while trying to find it. 
wish we had a mutation testing tool.